packages/cli/src/ceramic-logger-plugins.ts

```diff
@@ -54,29 +73,37 @@ export class LogToFiles {
     }
 
     /**
-     * Opens a filesystem stream and writes `message` to it
-     * @param filePath Full path of file to write to
-     * @param message Message to write to `filePath`
-     * @param writeFlag Specifies writing method (e.g. "a" for append, "w" for overwrite)
+     * Appends `message` to file `${basePath}${filePrefix}.log`
+     * @notice Rotates file if it is expired or of at least max size
+     * @param basePath Base directory for file
+     * @param filePrefix Prefix of file name to write to
+     * @param message Message to write to file
      */
-    private static async _writeStream (filePath: string, message: string, writeFlag: string): Promise<void> {
-        const fileExpired = await LogToFiles._isExpired(filePath)
-        fileExpired && await LogToFiles._rotate(filePath)
-        const stream = fs.createWriteStream(
-            filePath,
-            { flags: writeFlag }
-        )
-        stream.write(util.format(message) + '\n')
-        stream.end()
+    private static async _write (
+        blockedFiles: BlockedFiles,
+        basePath: string,
+        filePrefix: string,
+        message: string
+    ): Promise<void> {
+        const filePath = `${basePath}${filePrefix}.log`
+        await LogToFiles._shouldRotate(filePath) && await LogToFiles._rotate(filePath)
```

Comment:
hmm... stat-ing a file (which _shouldRotate does internally to get the file size and creation time) is a relatively expensive syscall.  Calling this on every single line logged may cause us problems down the road.  Instead of doing this inline to logging, can we have a periodic background task that checks if all log files need rotating and rotates any that do?

Reply:
Thanks for pointing this out. Alternatively we can track the file size/created time in some state and then rotate when it exceeds the threshold, that way we only have to do the stat when the process starts.

But I'd actually prefer using something like this: https://github.com/iccicci/rotating-file-stream
I believe that is what they are doing as well.

Reply:
Ooh yeah, if we can *not* have to write this ourselves that's definitely the way to go :sweat_smile:

Reply:
(I'm fine with deferring this to future optimization though)